File: 0647-palindromic-substrings/0647-palindromic-substrings.cpp
class Solution {
public:
    int expandindex(string s, int left,int right)
    {
        int count=0;
        while(left>=0 && right<s.length() && s[left]==s[right])
        {
            count++;
            left--;
            right++;
        }
        return count;
    }
    int countSubstrings(string s) {
        int count=0;
        int n=s.length();

        for(int centre=0;centre<n;centre++)
        {
            int oddkans=expandindex(s,centre,centre);
            count=count+oddkans;

            int evenkans=expandindex(s,centre,centre+1);
            count=count+evenkans;
        }
        return count;
    }
};

File: 0647-palindromic-substrings/0647-palindromic-substrings.java
class Solution {
    int count=0;
    public int countSubstrings(String s) {
       for (int start = 0; start < s.length(); start++) {
			expandRange(s, start, start);
			expandRange(s, start, start + 1);
		}
        return count;
    }
    public void expandRange(String str, int begin, int end) {
		while (begin >= 0 && end < str.length() && str.charAt(begin) == str.charAt(end)) {
            count++;
			begin--;
			end++;
		}
	}
}